### PR TITLE
Adding git-subtree-check.sh util script

### PR DESCRIPTION
**Pull request by furszy**
Script to check subtree directory consistency with its upstream repository.
Same script that we are using in the main PIVX core repo. Straight from BTC's repo.

### CODE REVIEW

File: devtools/git-subtree-check.sh [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      echo "referenced commit is present in any fetched remote."
+      echo ""
+      echo "DIR is the prefix within the repository to check."
+      echo "COMMIT is the commit to check, if it is not provided, HEAD will be used."
+      echo ""
+      echo "-r      Check that subtree commit is present in repository."
+      echo "        To do this check, fetch the subtreed remote first. Example:"
+      echo ""
+      echo "            git fetch https://github.com/bitcoin-core/secp256k1.git"
+      echo "            test/lint/git-subtree-check.sh -r src/secp256k1"
```
Comment (random-zebra) [inline devtools/git-subtree-check.sh]: similar to https://github.com/PIVX-Project/PIVX/pull/2473#discussion_r670526683, we can maybe do
```suggestion
      echo "            git fetch https://github.com/relic-toolkit/relic.git"
      echo "            devtools/git-subtree-check.sh -r contrib/relic"
```

which btw checks out here
```
contrib/relic in HEAD currently refers to tree 23de0f0a5c484c4eaa440e14836819a791ea3e13
contrib/relic in HEAD was last updated in commit 914ee5252e232a7308c9d38d9c259f344a38ee51 (tree 23de0f0a5c484c4eaa440e14836819a791ea3e13)
contrib/relic in HEAD was last updated to upstream commit 1885ae3b681c423c72b65ce1fe70910142cf941c (tree 23de0f0a5c484c4eaa440e14836819a791ea3e13)
GOOD
```

Comment (furszy) [reply devtools/git-subtree-check.sh]: yeah, will update it. Same as the other one.

**Comment by furszy** [comment]
Done, example updated + added the `gitHub-merge.py` script as well, so we start standardizing the repository workflow.

**Review by random-zebra — APPROVED** [review]
utACK 4ce70e6b12427ec1b92c50c30660dddeba8bb9a5

**Pull request merged**